contrib/completion/bash/docker

```diff
@@ -550,17 +550,18 @@ __docker_complete_nodes() {
 # output to the IDs or names of matching items. This setting takes
 # precedence over the environment setting.
 __docker_services() {
-	local fields='$2'  # default: service name only
-	[ "${DOCKER_COMPLETION_SHOW_SERVICE_IDS}" = yes ] && fields='$1,$2' # ID & name
+	local format='{{.Name}}'  # default: service name only
+	[ "${DOCKER_COMPLETION_SHOW_SERVICE_IDS}" = yes ] && format='{{.ID}},{{.Name}}' # ID & name
 
 	if [ "$1" = "--id" ] ; then
-		fields='$1' # IDs only
+		format='{{.ID}}' # IDs only
 		shift
 	elif [ "$1" = "--name" ] ; then
-		fields='$2' # names only
+		format='{{.Name}}' # names only
 		shift
 	fi
-        __docker_q service ls "$@" | awk "NR>1 {print $fields}"
+
+  __docker_q service ls -q --format "$format" --filter "name=$1"
```

Comment:
`--format` takes precedence over `-q`, so `-q` can be removed.

Reply:
@albers Not quite. The data sent from the docker daemon is truncated when using `-q`, therefore using `-q` is much faster. You can benchmark it on swarm clusters with a lot of services running. For example at my workplace:
```
$ docker service ls -q | wc -l
194

$ time docker service ls --format "{{json .}}" 2>/dev/null

real	0m1.254s
user	0m0.225s
sys	0m0.047s

$ time docker service ls -q --format "{{json .}}" 2>/dev/null

real	0m0.378s
user	0m0.093s
sys	0m0.026s
```

In this example I take the time only once, but I have benchmarked this with 1000 requests over time (each), and it seems to be that without `-q` it takes ~1.5s and with `-q` it takes ~0.4s. So for a tab completion script, I would say that `-q` is quite crucial.

Reply:
There is also one of our environments where we always get an error without `-q`:

```bash
$ docker service ls
Error response from daemon: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (4835595 vs. 4194304)

$ docker service ls -q | wc -l
780
```

Reply:
I did some testing and can confirm that while it does not change the output (in our examples), `--quiet` in fact positively impacts the performance. I did not expect this, thanks for pointing me to this.